apps/opik-backend/src/main/java/com/comet/opik/domain/evaluators/ManualEvaluationService.java

```
@@ -0,0 +1,149 @@
+}
+
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+@Slf4j
+class ManualEvaluationServiceImpl implements ManualEvaluationService {
+
+    private final @NonNull AutomationRuleEvaluatorService automationRuleEvaluatorService;
+    private final @NonNull OnlineScorePublisher onlineScorePublisher;
+
+    @Override
+    public Mono<ManualEvaluationResponse> evaluate(@NonNull ManualEvaluationRequest request, @NonNull UUID projectId,
+            @NonNull String workspaceId, @NonNull String userName) {
+
+        log.info(
+                "Starting manual evaluation for '{}' entities of type '{}' with '{}' rules in project '{}', workspace '{}'",
+                request.entityIds().size(), request.entityType(), request.ruleIds().size(), projectId, workspaceId);
+
+        return Mono.fromCallable(() -> {
+            Set<UUID> ruleIdSet = new HashSet<>(request.ruleIds());
+            List<?> rawRules = automationRuleEvaluatorService.findByIds(ruleIdSet, projectId, workspaceId);
+            @SuppressWarnings("unchecked")
+            List<AutomationRuleEvaluator<?>> rules = (List<AutomationRuleEvaluator<?>>) rawRules;
+
+            // Validate that all requested rules were found
+            if (rules.size() != request.ruleIds().size()) {
+                Set<UUID> foundRuleIds = rules.stream()
+                        .map(AutomationRuleEvaluator::getId)
+                        .collect(Collectors.toSet());
+
+                Set<UUID> missingRuleIds = ruleIdSet.stream()
+                        .filter(id -> !foundRuleIds.contains(id))
+                        .collect(Collectors.toSet());
+
+                log.warn("Some rule IDs were not found: '{}'", missingRuleIds);
+                throw new BadRequestException(
+                        "Automation rule(s) not found with ID(s): %s".formatted(
+                                missingRuleIds.stream().map(UUID::toString).collect(Collectors.joining(", "))));
+            }
```

Do we want to fail entire request if some rule is missing? Maybe perform evaluation with existing rules and add some message about missing ones?